Sources/App/Core/Dependencies/RedisClient.swift

```diff
@@ -0,0 +1,127 @@
+// Copyright Dave Verwer, Sven A. Schmidt, and other contributors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import NIOCore
+@preconcurrency import RediStack
+import Dependencies
+import DependenciesMacros
+
+
+@DependencyClient
+struct RedisClient {
+    var set: @Sendable (_ key: String, _ value: String?, Duration?) async -> Void
+    var get: @Sendable (_ key: String) async -> String?
```

Comment:
I've opted for the simplest interface, just taking `String` keys and values for starters. We can add overloads that take `T: Codable` which funnel strings through these methods if needed.